src/v/config/configuration.h

```diff
@@ -302,6 +302,8 @@ struct configuration final : public config_store {
     // enables rack aware replica assignment
     property<bool> enable_rack_awareness;
 
+    property<bool> storage_strict_data_init;
```

Comment:
nit: the properties aren't strictly sorted, but they're mostly grouped: would be nice to put this with other storage_ properties.